mm2src/lp_ordermatch.rs

```diff
@@ -463,95 +417,79 @@ impl TryFromBytes for Uuid {
 pub async fn process_peer_request(ctx: MmArc, request: OrdermatchRequest) -> Result<Option<Vec<u8>>, String> {
     println!("Got ordermatch request {:?}", request);
     match request {
-        OrdermatchRequest::GetOrderbook {
-            base,
-            rel,
-            asks_num,
-            bids_num,
-        } => process_get_orderbook_request(ctx, base, rel, asks_num, bids_num).await,
-        OrdermatchRequest::SyncPubkeyOrderbookState {
-            pubkey,
-            current_orders_trie_root,
-            expected_orders_trie_root,
-            pairs_trie_roots,
-        } => {
-            let response = process_sync_pubkey_orderbook_state(
-                ctx,
-                pubkey,
-                current_orders_trie_root,
-                expected_orders_trie_root,
-                pairs_trie_roots,
-            )
-            .await;
+        OrdermatchRequest::GetOrderbook { base, rel } => process_get_orderbook_request(ctx, base, rel).await,
+        OrdermatchRequest::SyncPubkeyOrderbookState { pubkey, trie_roots } => {
+            let response = process_sync_pubkey_orderbook_state(ctx, pubkey, trie_roots).await;
             response.map(|res| res.map(|r| encode_message(&r).expect("Serialization failed")))
         },
     }
 }
 
-async fn process_get_orderbook_request(
-    ctx: MmArc,
-    base: String,
-    rel: String,
-    asks_num: Option<usize>,
-    bids_num: Option<usize>,
-) -> Result<Option<Vec<u8>>, String> {
-    /*
-    enum PriceOrdering {
-        LowestToHighest,
-        HighestToLowest,
-    }
-    fn get_n_orders(
+#[allow(dead_code)]
+type TrieProof = Vec<Vec<u8>>;
+
+#[derive(Debug, Deserialize, Serialize)]
+struct GetOrderbookPubkeyItem {
+    /// Timestamp of the latest keep alive message received.
+    last_keep_alive: u64,
```

Comment:
I didn't have time to add `last_signed_message`.

Reply:
As I can see pair trie root hash was not added to the `MakerOrderCreated` and other order related messages. It is essential to have it along with `last_signed_message`. If we don't have it we have no proof that specific orders do really belong to the pubkey.

The node will notice that it has wrong data upon keep alive, but we should also keep in mind that in the future we might need to request orderbook without subscription to specific topic.

Reply:
It might be also nice to add `last_signed_message` to sync pubkey state response. There might be a case that root changes between keep alive is received and node received keep alive sends request to sync orderbook. This way the node will node that new state not matching keep alive is valid.